src/Akeneo/CouplingDetector/CouplingDetector.php

```diff
@@ -39,16 +41,30 @@ public function __construct(NodeParserResolver $resolver, RuleChecker $checker)
     /**
      * Detect the coupling errors of the nodes that are found among a set of rules.
      *
-     * @param Finder          $finder
-     * @param RuleInterface[] $rules
+     * @param Finder               $finder
+     * @param RuleInterface[]      $rules
+     * @param ExclusionInterface[] $exclusions
      *
      * @return ViolationInterface[]
      */
-    public function detect(Finder $finder, array $rules)
+    public function detect(Finder $finder, array $rules, array $exclusions = [])
     {
         $nodes = $this->parseNodes($finder);
-        $violations = [];
+        $violations = $this->detectViolations($rules, $nodes);
+        $violations = $this->filterViolations($violations, $exclusions);
```

Comment:
it's a little hackish too detect all violations and then remove the one that come from something that has been excluded
I think we should check that from the RuleChecker, and exclude the tokens of the nodes that are excluded